src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -463,6 +466,77 @@ public Double getPredefinedResult(LfSensitivityFactor lfSensitivityFactor) {
+                                Collection<Pair<LfBus, LccConverterStation>> lccs, Collection<Pair<LfBus, LfVscConverterStationImpl>> vscs) {
+        AbstractLfBus bus = (AbstractLfBus) lfNetwork.getBusById(station.getTerminal().getBusView().getBus().getId());
+        if (bus == null) {
+            return;
+        }
+        if (station instanceof LccConverterStation) {
+            lccs.add(Pair.of(bus, (LccConverterStation) station));
+        } else if (station instanceof VscConverterStation) {
+            vscs.add(Pair.of(bus, bus.getGenerators().stream().filter(LfVscConverterStationImpl.class::isInstance)
+                .map(LfVscConverterStationImpl.class::cast)
+                .filter(lfVscConverterStation -> lfVscConverterStation.getId().equals(station.getId()))
+                .findFirst().orElseThrow()));
+        }
+    }
+
+    private Map<LfBus, BusState> applyHvdcContingency(Network network, LfNetwork lfNetwork, PropagatedContingency contingency) {
+        Collection<Pair<LfBus, LccConverterStation>> lccs = new HashSet<>();
+        Collection<Pair<LfBus, LfVscConverterStationImpl>> vscs = new HashSet<>();
+        for (String hvdcId : contingency.getHvdcIdsToOpen()) {
+            HvdcLine hvdcLine = network.getHvdcLine(hvdcId);
+            processStation(lfNetwork, hvdcLine.getConverterStation1(), lccs, vscs);
+            processStation(lfNetwork, hvdcLine.getConverterStation2(), lccs, vscs);
+        }
+        Collection<LfBus> busesToSave = new HashSet<>();
+        busesToSave.addAll(lccs.stream().map(Pair::getKey).collect(Collectors.toSet()));
+        busesToSave.addAll(vscs.stream().map(Pair::getKey).collect(Collectors.toSet()));
+        Map<LfBus, BusState> busStates = BusState.createBusStates(busesToSave);
+
+        for (Pair<LfBus, LccConverterStation> busAndlcc : lccs) {
+            LfBus bus = busAndlcc.getKey();
+            LccConverterStation lcc = busAndlcc.getValue();
+            HvdcLine line = lcc.getHvdcLine();
+            // The active power setpoint is always positive.
+            // If the converter station is at side 1 and is rectifier, p should be positive.
+            // If the converter station is at side 1 and is inverter, p should be negative.
+            // If the converter station is at side 2 and is rectifier, p should be positive.
+            // If the converter station is at side 2 and is inverter, p should be negative.
+            boolean isConverterStationRectifier = HvdcConverterStations.isRectifier(lcc);
+            double pCs = (isConverterStationRectifier ? 1 : -1) * line.getActivePowerSetpoint() *
+                (1 + (isConverterStationRectifier ? 1 : -1) * lcc.getLossFactor() / 100); // A LCC station has active losses.
+            double qCs = Math.abs(pCs * Math.tan(Math.acos(lcc.getPowerFactor()))); // A LCC station always consumes reactive power.
+            bus.setLoadTargetP(bus.getLoadTargetP() - pCs / PerUnit.SB);
```

Comment:
Are you sure that it works if you have a load or a shunt in the same connexion bus than your LCC converter station?

Reply:
Okay I get it... mmmmm... could you add a FIXME for me later if we keep this like that?